core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2046,7 +2046,7 @@ class ReplicaManager(val config: KafkaConfig,
 
   def getLogConfig(topicPartition: TopicPartition): Option[LogConfig] = localLog(topicPartition).map(_.config)
 
-  def getMagic(topicPartition: TopicPartition): Option[Byte] = getLogConfig(topicPartition).map(_.recordVersion.value)
+  def getMagic(topicPartition: TopicPartition): Option[Byte] = getLogConfig(topicPartition).map(_ => RecordVersion.V2.value)
```

Comment:
Do we still need this method?

Reply:
Thanks for review and suggestion. I try to remove `ReplicaManager#getMagic` and it looks like some related logic on this path can be removed too. Can I also remove them? Thanks.

* https://github.com/apache/kafka/blob/fe56fc98fa736c79c9dcbb1f64f810065161a1cc/clients/src/main/java/org/apache/kafka/common/record/AbstractRecords.java#L93C23-L118
* https://github.com/apache/kafka/blob/fe56fc98fa736c79c9dcbb1f64f810065161a1cc/clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java#L94-L119

Reply:
Perhaps we can leave the changes to the records for a separate PR. One thing we have to be careful about is that old record formats may exist on disk - so the functionality to handle that needs to remain.